src/backend/api/src/result.rs

```diff
@@ -0,0 +1,62 @@
+use candid::{CandidType, Deserialize};
+
+#[derive(Debug, CandidType, Deserialize)]
+pub enum ApiResult<T = ()> {
+    #[serde(rename = "ok")]
+    Ok(T),
+
+    #[serde(rename = "err")]
+    Err(ApiError),
+}
+
+#[derive(Debug, Clone, CandidType, Deserialize, PartialEq, Eq)]
+pub struct ApiError {
+    code: u16,
+    message: String,
+}
+
+impl ApiError {
+    pub fn invalid_argument(message: &str) -> Self {
```

Comment:
Using REST-inspired error codes means we don't need to design new ones. I try to avoid panics and traps as much as possible so that we can hook into the `Result` type and log any `Err` types that we come across.

The main area I'm concerned about with trapping is the stable structures because they have many reasons to panic internally. Unit tests are making sure we're safe from canister controlled data causing a panic, but once we have input coming from outside, we may want add data validation that we can return a `Err` from and prevent stable structures from panicking.